opentelemetry-sdk/src/resource/env.rs

```diff
@@ -88,7 +88,10 @@ impl ResourceDetector for SdkProvidedResourceDetector {
                         .get(Key::new("service.name"))
                         .map(|v| v.to_string())
                         .filter(|s| !s.is_empty())
-                        .unwrap_or_else(|| "unknown_service".to_string())
+                        .unwrap_or_else(|| match option_env!("CARGO_BIN_NAME") {
+                            Some(s) => s.to_string(),
```

Comment:
Also I apologize for glossing over this before but the idea is to use `env!()` which should always produce a value although https://doc.rust-lang.org/std/macro.env.html would likely serve us best just in case. 
This will grab it at build time whereas the current configuration grabs at runtime.

Reply:
Sorry if I am misunderstanding what you say, but won't using `env!` in a library always fail? Where as using `option_env!` allows for it to compile as a `lib` and will resolve properly when being used as part of a `bin`. 

Please correct me if I am missing or misunderstanding something here.

Reply:
I think my review was a delayed, I hadn't seen djc's response 😓